### PR TITLE
test: add unit test for flag trigger

### PR DESCRIPTION
**Pull request by kakcy**
This pull request primarily includes adding unit tests for FlagTrigger and changes to the Bucketeer project's "pkg/feature/api" package. This change focuses on refactoring the `FeatureService` structure and related methods in `api.go` and `flag_trigger.go` to use the new `v2fs` package for feature storage. Additionally, error handling has been simplified for several methods, and the "generateTriggerURL" method no longer returns an error.

Here are the most important changes:

Add unit test for flag trigger.

Refactoring to use `v2fs` package:

* [`pkg/feature/api/api.go`](diffhunk://#diff-982bed6b45a5128e1cac0183ec37fca993ed8174a115201c70c168f63f8e9043R27-R29): Imported the `v2fs` package and its `FeatureStorage` and `FlagTriggerStorage` types. These were then added as fields to the `FeatureService` struct, replacing direct usage of the `mysqlClient`. The `NewFeatureService` function was also updated to initialize these new fields. [[1]](diffhunk://#diff-982bed6b45a5128e1cac0183ec37fca993ed8174a115201c70c168f63f8e9043R27-R29) [[2]](diffhunk://#diff-982bed6b45a5128e1cac0183ec37fca993ed8174a115201c70c168f63f8e9043R57-R58) [[3]](diffhunk://#diff-982bed6b45a5128e1cac0183ec37fca993ed8174a115201c70c168f63f8e9043R91-R92)
* [`pkg/feature/api/api_test.go`](diffhunk://#diff-b900941517f1d0f0d7f739282ddc32ac4cbdb33f4bd2291b1a00d4d7d10912dbR26-R27): Similar changes were made in the test file, where mock instances of `FeatureStorage` and `FlagTriggerStorage` are now created and used in the `createFeatureService` and `createFeatureServiceNew` functions. [[1]](diffhunk://#diff-b900941517f1d0f0d7f739282ddc32ac4cbdb33f4bd2291b1a00d4d7d10912dbR26-R27) [[2]](diffhunk://#diff-b900941517f1d0f0d7f739282ddc32ac4cbdb33f4bd2291b1a00d4d7d10912dbR125-R126) [[3]](diffhunk://#diff-b900941517f1d0f0d7f739282ddc32ac4cbdb33f4bd2291b1a00d4d7d10912dbR153-R154)

Simplification of error handling:

* [`pkg/feature/api/flag_trigger.go`](diffhunk://#diff-33bed37b43315e58942e192c4c1c99954795817fae734470f6a5ea5c8a218706L142-R142): In several methods, the error handling for `generateTriggerURL` has been removed, as this function no longer returns errors. Instead, the URL is directly assigned to `triggerURL`. Additionally, error handling in `ResetFlagTrigger` and `DeleteFlagTrigger` has been simplified to return a single error status. [[1]](diffhunk://#diff-33bed37b43315e58942e192c4c1c99954795817fae734470f6a5ea5c8a218706L142-R142) [[2]](diffhunk://#diff-33bed37b43315e58942e192c4c1c99954795817fae734470f6a5ea5c8a218706L542-R544) [[3]](diffhunk://#diff-33bed37b43315e58942e192c4c1c99954795817fae734470f6a5ea5c8a218706L645-R645) [[4]](diffhunk://#diff-33bed37b43315e58942e192c4c1c99954795817fae734470f6a5ea5c8a218706L669-R669) [[5]](diffhunk://#diff-33bed37b43315e58942e192c4c1c99954795817fae734470f6a5ea5c8a218706L687-R687) [[6]](diffhunk://#diff-33bed37b43315e58942e192c4c1c99954795817fae734470f6a5ea5c8a218706L755-R748) [[7]](diffhunk://#diff-33bed37b43315e58942e192c4c1c99954795817fae734470f6a5ea5c8a218706L827-R813) [[8]](diffhunk://#diff-33bed37b43315e58942e192c4c1c99954795817fae734470f6a5ea5c8a218706L860-R845) [[9]](diffhunk://#diff-33bed37b43315e58942e192c4c1c99954795817fae734470f6a5ea5c8a218706L1061-R1049)

Refactoring of `FeatureService` methods:

* [`pkg/feature/api/flag_trigger.go`](diffhunk://#diff-33bed37b43315e58942e192c4c1c99954795817fae734470f6a5ea5c8a218706L669-R669): All methods in this file that previously created new instances of `FlagTriggerStorage` or `FeatureStorage` now use the instances stored in the `FeatureService` struct. This includes the `GetFlagTrigger`, `FlagTriggerWebhook`, and `generateTriggerURL` methods. [[1]](diffhunk://#diff-33bed37b43315e58942e192c4c1c99954795817fae734470f6a5ea5c8a218706L669-R669) [[2]](diffhunk://#diff-33bed37b43315e58942e192c4c1c99954795817fae734470f6a5ea5c8a218706L827-R813) [[3]](diffhunk://#diff-33bed37b43315e58942e192c4c1c99954795817fae734470f6a5ea5c8a218706L860-R845)

### CODE REVIEW

File: pkg/feature/api/flag_trigger_test.go [hunk 1]
```diff
@@ -46,15 +827,15 @@ func TestFeatureServiceGenerateTriggerURL(t *testing.T) {
 		t.Errorf("GenerateToken() error = %v", err)
 	}
 	t.Logf("GenerateToken() token = %v", trigger.Token)
-	triggerURL, err := featureService.generateTriggerURL(context.Background(), trigger.Token, false)
+	triggerURL := featureService.generateTriggerURL(context.Background(), trigger.Token, false)
 	if err != nil {
 		t.Errorf("generateTriggerURL() [full] error = %v", err)
 	}
```
Comment (kentakozuka) [inline pkg/feature/api/flag_trigger_test.go:833]:
```suggestion
```

File: pkg/feature/api/flag_trigger_test.go [hunk 2]
```diff
@@ -46,15 +827,15 @@ func TestFeatureServiceGenerateTriggerURL(t *testing.T) {
...
 	if triggerURL == "" {
 		t.Errorf("generateTriggerURL() [full] triggerURL is empty")
 	}
 	t.Logf("generateTriggerURL() [full] triggerURL = %v", triggerURL)
-	triggerURL, err = featureService.generateTriggerURL(context.Background(), trigger.Token, true)
+	triggerURL = featureService.generateTriggerURL(context.Background(), trigger.Token, true)
 	if err != nil {
 		t.Errorf("generateTriggerURL() [masked] error = %v", err)
 	}
```
Comment (kentakozuka) [inline pkg/feature/api/flag_trigger_test.go:841]:
```suggestion
```

File: pkg/feature/api/flag_trigger_test.go [hunk 3]
```diff
@@ -17,14 +17,795 @@ package api
...
 
+func TestCreateFlagTrigger(t *testing.T) {
+	t.Parallel()
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+
+	ctx := createContextWithToken()
+	ctx = metadata.NewIncomingContext(ctx, metadata.MD{
+		"accept-language": []string{"ja"},
+	})
```
Comment (kentakozuka) [inline pkg/feature/api/flag_trigger_test.go:47]:
```suggestion
	ctx = metadata.NewIncomingContext(
		createContextWithToken(),
		metadata.MD{"accept-language": []string{"ja"}},
	)
```

**Review by kentakozuka — CHANGES_REQUESTED** [review]
Thanks!
left some suggestions :)

**Comment by kakcy** [comment]
@kentakozuka 
Thank you for your suggestions!
I adopted it because it was necessary.
https://github.com/bucketeer-io/bucketeer/pull/773/commits/dbad210cda8cbe0adde84f32688a1ecdddc005be
https://github.com/bucketeer-io/bucketeer/pull/773/commits/f80df540e9a3da3d034d78b262848f9ac9f43661

**Review by kentakozuka — APPROVED** [review]
LGTM 👍

**Comment by kakcy** [comment]
Thank you for the reviews!

**Pull request merged**